Library/Homebrew/extend/pathname.rb

```diff
@@ -131,6 +131,12 @@ def write(content, *open_args)
     open("w", *open_args) { |f| f.write(content) }
   end
 
+  # Only appends to a file that is already created.
+  def write_append(content, *open_args)
+    raise "Cannot append file that doesn't exist: #{self}" unless exist?
+    open("a", *open_args) { |f| f.puts(content) }
```

Comment:
> I think it should be f.write instead of f.puts here, as it's surprise to silently append newline. The caller should explicitly pass the newline character if needed.

I think `f.puts` is consistent with current usage and expectation, but I guess I don't have a particularly strong objection if nobody else does.